perception/autoware_multi_object_tracker/lib/tracker/model/vehicle_tracker.cpp

```diff
@@ -260,8 +260,29 @@ bool VehicleTracker::measure(
 
   // update object
   types::DynamicObject updating_object = in_object;
+  // turn 180 deg if the updating object heads opposite direction
+  {
+    const double this_yaw = motion_model_.getStateElement(IDX::YAW);
+    const double updating_yaw = tf2::getYaw(updating_object.pose.orientation);
+    double yaw_diff = updating_yaw - this_yaw;
+    while (yaw_diff > M_PI) yaw_diff -= 2 * M_PI;
+    while (yaw_diff < -M_PI) yaw_diff += 2 * M_PI;
+    if (std::abs(yaw_diff) > M_PI_2) {
+      tf2::Quaternion q;
+      q.setRPY(0, 0, updating_yaw + M_PI);
+      updating_object.pose.orientation = tf2::toMsg(q);
+      updating_object.anchor_point.x = -updating_object.anchor_point.x;
+      updating_object.anchor_point.y = -updating_object.anchor_point.y;
```

Comment:
Currently we can keep it. If we want to use this in 3D rotation, we may need something like:

```
tf2::Quaternion q_orig;
tf2::fromMsg(updating_object.pose.orientation, q_orig);

tf2::Quaternion q_flip;
q_flip.setRotation(tf2::Vector3(0, 0, 1), M_PI);  // π rad = 180°

tf2::Quaternion q_new = q_orig * q_flip;  // locally rotate 180deg
q_new.normalize();
```

IMO this option should not activated for computation resource issue.